File: src/main/java/fi/quanfoxes/assembler/Instructions.java
package fi.quanfoxes.assembler;

public class Instructions {
    private StringBuilder builder = new StringBuilder();
    private Reference reference;

    public Instructions comment(String comment) {
        builder = builder.append("; ").append(comment).append("\n");
        return this;
    }

    public Instructions label(String name) {
        builder = builder.append(name).append(":\n");
        return this;
    }

    public Instructions append(String raw) {
        builder = builder.append(raw).append("\n");
        return this;
    }

    public Instructions append(String format, Object... args) {
        builder = builder.append(String.format(format, args)).append("\n");
        return this;
    }

    public Instructions append(Instruction instruction) {
        builder = builder.append(instruction).append("\n");
        return this;
    }

    public Instructions append(Instructions...instructions) {
        for (Instructions i : instructions) {
            builder = builder.append(i);
        }

        return this;
    }

    public Instructions setReference(Reference reference) {
        this.reference = reference;
        return this;
    }
    
    public Reference getReference() {
        return reference;
    }

    @Override
    public String toString() {
        return builder.toString();
    }
}

File: Zigzag/Assembler/Instructions.cs
using System.Text;

public class Instructions
{
	private StringBuilder Builder = new StringBuilder();
	public Reference Reference { get; set; }

	public static Instructions GetReference(Reference reference)
	{
		return new Instructions().SetReference(reference);
	}

	public Instructions Comment(string comment)
	{
		Builder = Builder.Append("; ").Append(comment).Append("\n");
		return this;
	}

	public Instructions Label(string name)
	{
		Builder = Builder.Append(name).Append(":\n");
		return this;
	}

	public Instructions Append(string raw)
	{
		Builder = Builder.Append(raw).Append("\n");
		return this;
	}

	public Instructions Append(string format, params object[] args)
	{
		Builder = Builder.Append(string.Format(format, args)).Append("\n");
		return this;
	}

	public Instructions Append(Instruction instruction)
	{
		Builder = Builder.Append(instruction).Append("\n");
		return this;
	}

	public Instructions Append(params Instructions[] instructions)
	{
		foreach (Instructions i in instructions)
		{
			Builder = Builder.Append(i);
		}

		return this;
	}

	public void Break()
	{
		Builder = Builder.Append('\n');
	}

	public Instructions SetReference(Reference reference)
	{
		Reference = reference;
		return this;
	}

	public override string ToString()
	{
		return Builder.ToString();
	}
}